Add feeder to crawl all MediaWiki pages

MrSerth opened the pull request:
This PR adds a feeder crawling all pages from the specified MediaWiki instance. It may be used to initialize a new instance of DBpedia (with an empty MySQL cache and an empty Virtuoso) or after the RCStream processing was stopped for any reason. In the second case, it differs from the UnmodifiedFeeder as this new AllPagesFeeder will also add new pages where the UnmodifiedFeeder is unable to handle that case.

jimkont commented on live/pom.xml:
```diff
@@ -180,6 +180,12 @@
         </dependency>
 
         <dependency>
+            <groupId>org.json</groupId>
```
we already use the jackson library, if we do not need this one specifically I suggest we switch to jackson

jimkont commented on live/src/main/java/org/dbpedia/extraction/live/feeder/AllPagesFeeder.java:
```diff
@@ -0,0 +1,99 @@
+
+    private boolean isFinished = false;
+    private String query_base =
+            "?action=query&format=json&list=allpages&aplimit=500&apnamespace=%d&continue=%s&apcontinue=%s";
```
In general, scala has very good support for XML with type safety, I suggest we switch to XML here and be consistent with the rest of the Api calls

jimkont commented:
Looks very good except a minor comment for json/XML

MrSerth commented:
I refactored the AllPagesFeeder to use the WikiApi and I switched to XML. In addition to that, I aligned the indention in the WikiApi (to make all methods similar in their indention).

jimkont commented on live/src/main/java/org/dbpedia/extraction/live/feeder/AllPagesFeeder.java:
```diff
@@ -0,0 +1,99 @@
+        if (!isFinished) {
+            List<Node> pageList = queryAllPagesAPI();
+            for (Node page : pageList) {
+                //queue.add(new LiveQueueItem(Long.parseLong(page.$bslash$at("pageid")), page.$bslash$at("title"), DateUtil.transformToUTC(new Date()), false, ""));
```
Looks like a typo :)

MrSerth replied:
Thanks. I was testing the Feeder to run on the English Wikipedia to ensure we don't have any loop (and obviously, I forgot to revert that).

jimkont commented:
+1
